app/knight.py

```diff
@@ -0,0 +1,33 @@
+from __future__ import annotations
+
+
+class Knight:
+    def __init__(self, knight: dict) -> None:
+        self.knight = knight
+
+    def preparation(self) -> None:
+        self.name = self.knight["name"]

```

Comment:
The `name` attribute is not declared in the `__init__` method. It's a good practice to declare all attributes in the `__init__` method. This is a code quality issue.